Improve TL;DR documentation for starters (#215)
decidim/decidim-bulletin-board: bulletin_board/server/README.md

The Bulletin Board depends on the existence of a Decidim installation.

> ## Runining on Docker

Reviewer: ```suggestion
## Running on Docker
```

You can run this application conveniently packaged with docker compose.

You can use the existing [docker-composer.yml](docker-composer.yml) as start point.
It requires you to define a `.env` file with the following ENV variables:

```
SECRET_KEY_BASE=-your-rails-application-secret-
DATABASE_URL=postgres://USERNAME:PASSWOR@DATABASE/BULLETIN_DB
IDENTIFICATION_PRIVATE_KEY=-private-key-for-this-bulletin-board

```

#### Notes:

> - Generate `SECRET_KEY_BASE` with `rails secret`

Reviewer: ```suggestion
- Generate `SECRET_KEY_BASE` with `bin/rails secret`
```

- Generate `IDENTIFICATION_PRIVATE_KEY` with `bin/rails client:generate_identification_private_key`

### Configuration steps in this server (bulletin app):

> - Initialize the database, database is not included in this docker compose file but it should be pretty straithforward.

Reviewer: ```suggestion
- Initialize the database, database is not included in this docker compose file but it should be pretty straightforward.
```

  In order to initialize the database you need to run the comand `bin/rails db:create db:migrate`

- In you Decidim server (the one you want to connect with this one), execute
	`bin/rails decidim_elections:generate_identification_keys` (save the private key in you decidim environment)
	copy the public key as is presented (url encoded) and, now in this server, execute:

	`bin/rails 'client:add_authority[Authority name, public-key]'`

	Note that "Authority name" can be anything as long is the same in Decidim than here.

	**Important**: This step throws also an `API KEY` secret that needs to be configured in Decidim too.

#### Running the app with docker-compose

```bash
docker-compose up -d
```

Note that you still need to configure some proxy around it.

> For instance a simple Nginx conf can be:

Reviewer: ```suggestion
For instance, a simple Nginx conf for a server that's configured at bulletin-board.example.org can be:
```

```
server {

>     server_name bulletin.audit.platoniq.net;

Reviewer: ```suggestion
    server_name bulletin-board.example.org;
```

    client_max_body_size 32M;
    listen 80 ;
    listen [::]:80 ;

>     #listen [::]:443 ssl ipv6only=on; # managed by Certbot

Reviewer: Let's remove the commented lines of SSL configuration

    #listen 443 ssl; # managed by Certbot
    #ssl_certificate ...
    #ssl_certificate_key ...
    #include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    #ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

    location / {
        proxy_pass http://127.0.0.1:8000;
   }
}

```

> Note: you should add SSL configuration, for instance with [letsencrypt](https://certbot.eff.org/lets-encrypt)

Reviewer: ```suggestion
Note: you should add SSL configuration, for instance with [Let's Encrypt](https://certbot.eff.org/lets-encrypt)
```

> This server is ready, you can access the hompage either at https://localhost:8000 or https:/you-server.com depending on your final configuration.

Reviewer: ```suggestion
This server is ready, you can access the hompage either at http://bulletin-board.example.org or https://bulletin-board.example.org depending on your final configuration.
```

Author: Actually, the localhost is relevant here, more like to change it to http://localhost:8000 or http(s)//bulletin-board.example.org

```

In there, you can execute the `bin/rails c` to access the rails console, or any other command.

It is also possible to directly execut rails (or rake) comands:

```
docker-compose exec bulletin bin/rails db:migrate
```

**Important**, your application needs to be up and running for this to work (aka: `docker-compose up`)

### Configuration steps in Decidim server

This steps needs to be performed in the Decidim instance that wants to use the elections module.

Once you have this server configured, you will need the public key corresponding to the private key used in the
env var `IDENTIFICATION_PRIVATE_KEY`, simply use a browser and navigate to the homepage of the application, it will show the public key.

- Copy the public key to the env var `BULLETIN_BOARD_PUBLIC_KEY`
- Copy the `API KEY` obtained previously into the env var `BULLETIN_BOARD_API_KEY`
- Ensure you have the env var `AUTHORITY_PRIVATE_KEY` with the **private key** generate with the command `bin/rails decidim_elections:generate_identification_keys`
- Ensure that the `Authority name` is the same in both places (in Decidim needs to be declarated in the env var `AUTHORITY_NAME`)

> - Be sure to point the env var `BULLETIN_BOARD_SERVER` to the bulletin server API path (that is usually https://yout-server.com/api)

Reviewer: ```suggestion
- Be sure to point the env var `BULLETIN_BOARD_SERVER` to the bulletin server API path (that is usually http://bulletin-board.example.org/api)
```
